nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashMapShufflingAdder.java

```diff
@@ -211,31 +159,30 @@ public void visitInsn(int opcode) {
             hasNextProp.signature = signature;
             hasNextProp.exceptions = exceptions;
 
-            MethodVisitor original = super.visitMethod(access, "original_hasNext", desc, signature, exceptions);
-
-            return original;
+            return super.visitMethod(access, "original_hasNext", desc, signature, exceptions);
         }
-        if ("nextNode".equals(name)) {
-            nextNodeProp.access = access;
-            nextNodeProp.name = name;
-            nextNodeProp.desc = desc;
-            nextNodeProp.signature = signature;
-            nextNodeProp.exceptions = exceptions;
-
-            MethodVisitor original = super.visitMethod(access, "original_nextNode", desc, signature, exceptions);
-
-            return original;
-        }
-        if ("nextEntry".equals(name)) {
-            nextNodeProp.access = access;
-            nextNodeProp.name = name;
-            nextNodeProp.desc = desc;
-            nextNodeProp.signature = signature;
-            nextNodeProp.exceptions = exceptions;
-
-            MethodVisitor original = super.visitMethod(access, "original_nextEntry", desc, signature, exceptions);
-
-            return original;
+        if (name.equals("next" + type)) {
+            nextTypeProp.access = access;
+            nextTypeProp.name = name;
+            nextTypeProp.desc = desc;
+            nextTypeProp.signature = signature;
+            nextTypeProp.exceptions = exceptions;
+
+            if (baseName.equals("java/util/WeakHashMap")) {
```

Comment:
@azy2 This is worth fixing. (be robust to NPE) I am going to disable the other string literal checks/warnings.